activejob/lib/active_job/test_helper.rb

```diff
@@ -7,8 +8,16 @@ module TestHelper
 
     included do
       def before_setup
-        @old_queue_adapter = queue_adapter
-        ActiveJob::Base.queue_adapter = :test
+        test_adapter = ActiveJob::QueueAdapters::TestAdapter.new
+
+        @old_queue_adapters = (ActiveJob::Base.subclasses << ActiveJob::Base).select do |klass|
+          klass.respond_to?(:queue_adapter)
```

Comment:
Every `ActiveJob::Base` subclass will respond to `:queue_adapter` since it's a singleton method on `Base`